lib/msf/core/payload/stager.rb

```diff
@@ -156,7 +156,14 @@ def handle_connection(conn, opts={})
     # If the stage should be sent over the client connection that is
     # established (which is the default), then go ahead and transmit it.
     if (stage_over_connection?)
-      p = generate_stage
+      uuid_raw = conn.get_once(16, 1)
```

Comment:
So what happens in the case where a listener is, say, `reverse_tcp` but the stage that's executing is `reverse_tcp_uuid` ? Is the plan to just make the assumption that these will remain in sync?